### ISSUE TITLE
scipy.cluster.hierachy.dendrogram raises an error when used with numpy.ndarray labels

### ISSUE THREAD

**Issue by jnooree** [scipy/scipy#12444]
#### Reproducing code example:

```
from scipy.cluster.hierachy import dendrogram

dn = dendrogram(some_valid_ndarray, labels=np.arange(132))
```

#### Error message:

```
Traceback (most recent call last):
  File "utils/test.py", line 3, in <module>
    tani_corr(["200510_vae_mod_average_generated.mol2"], cwd="/home/jnooree/chemvae_pytorch/generated")
  File "/home/jnooree/chemvae_pytorch/mol_stats.py", line 47, in tani_corr
    dn = hier.dendrogram(link_mat,
  File "/home/jnooree/anaconda3/envs/pytorch/lib/python3.8/site-packages/scipy/cluster/hierarchy.py", line 3277, in dendrogram
    if labels and Z.shape[0] + 1 != len(labels):
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

#### Scipy/Numpy/Python version information:

```
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
 ##1.5.0 1.18.1 sys.version_info(major=3, minor=8, micro=3, releaselevel='final', serial=0)
```

#### Personal comments
I think this bug is from `numpy`, which now does not support direct "truthy" evaluation of `ndarray`. I don't know the best way to solve this, but just fixing the line like
```
if labels is not None and Z.shape[0] + 1 != len(labels):
```
solved the problem for me.

**Comment by AtsushiSakai** [comment]
@jnooree Hi. Thank you for reporting. Yes, that is a bug included in v1.5.0 and it was already fixed in #12421. It will be fixed in v1.5.1 or later.

**Comment by jnooree** [comment]
That’s a good news. Thank you for responding!
